src/github/api/client.rs

```diff
@@ -12,6 +15,31 @@ use crate::github::api::operations::{MergeError, merge_branches, set_branch_to_c
 use crate::github::{CommitSha, GithubRepoName, PullRequest, PullRequestNumber};
 use crate::utils::timing::measure_network_request;
 
+pub async fn auto_retry_github_request<T, F, Fut>(
+    request_name: &str,
+    f: F,
+) -> Result<T, octocrab::Error>
+where
+    F: Fn() -> Fut + Copy,
+    Fut: std::future::Future<Output = Result<T, octocrab::Error>>,
+{
+    let mut res = measure_network_request(request_name, f).await;
+    if let Err(octocrab::Error::Service { source, .. }) = &res {
```

Comment:
Hmm, I think that this is too ugly in the end. But mainly, I realized that if octocrab changes its internal representation of the errors, then this will silently stop working, and that's not great.

Let's just use the tokio timeout then (but still keep the octocrab timeout setting, that's good practice IMO), at least we will have more control over it.

I would generalize `measure_network_request`, rename it to something like `perform_network_request`, and pass it a timeout and a number of retries that it should perform when a timeout is encountered. By doing the timeouts in that function, we will be able to provide better log messages about what happens, because currently the log would just contain something like "operation started", and the "operation finished" part would be missing.